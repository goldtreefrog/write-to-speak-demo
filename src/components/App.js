import React, { Component } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "normalize.css";
import "./styles/App.css";
import Header from "./header";
import Footer from "./footer";
import Write from "./write";
import Home from "./home";
import Talk from "./talk";
import Edit from "./edit";
import LoginPage from "./login-page";
import RegisterForm from "./register-form";
import { refreshAuthToken } from "./../store/actions/auth.js";
import NotFound from "./not-found";

// Use named export for unconnected component (for tests)
export class App extends Component {
  componentDidUpdate(prevProps) {
    if (!prevProps.loggedIn && this.props.loggedIn) {
      // When we are logged in, refresh the auth token periodically
      this.startPeriodicRefresh();
    } else if (prevProps.loggedIn && !this.props.loggedIn) {
      // Stop refreshing when we log out
      this.stopPeriodicRefresh();
    }
  }

  componentWillUnmount() {
    this.stopPeriodicRefresh();
  }

  startPeriodicRefresh() {
    this.refreshInterval = setInterval(
      () => this.props.dispatch(refreshAuthToken()),
      60 * 60 * 1000 // One hour
    );
  }

  stopPeriodicRefresh() {
    if (!this.refreshInterval) {
      return;
    }

    clearInterval(this.refreshInterval);
  }

  render() {
    return (
      <Router>
        <React.Fragment>
          <div className="App">
            <Header />
            <main>
              <Switch>
                <Route path="/" component={Home} exact />
                <Route
                  exact
                  path="/write"
                  render={() =>
                    this.props.writing.isEditing &&
                    !this.props.writing.editingPage === "/write" ? (
                      <Redirect to={this.props.writing.editingPage} />
                    ) : (
                      <Write />
                    )
                  }
                />
                <Route path="/talk" component={Talk} exact />
                <Route
                  exact
                  path="/edit"
                  render={() =>
                    this.props.writing.isEditing &&
                    !this.props.writing.editingPage === "/edit" ? (
                      <Redirect to={this.props.writing.editingPage} />
                    ) : (
                      <Edit />
                    )
                  }
                />
                <Route path="/login" component={LoginPage} exact />
                <Route path="/register" component={RegisterForm} exact />
                <Route path="/logout" component={Home} exact />
                <Route component={NotFound} />
              </Switch>
            </main>
          </div>
          <Footer />
        </React.Fragment>
      </Router>
    );
  }
}

const mapStateToProps = state => {
  return {
    snippets: state.snippets,
    writing: state.writing,
    hasAuthToken: state.auth.authToken !== null,
    loggedIn: state.auth.currentUser !== null
  };
};

// Use default export for the connected component (for app)
export default connect(mapStateToProps)(App);
