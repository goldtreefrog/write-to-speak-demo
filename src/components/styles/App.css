.App {
  text-align: center;
  width: 100%;
}

button {
  border-radius: 10px;
}
