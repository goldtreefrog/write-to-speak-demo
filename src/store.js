import { createStore } from "redux";

const store = createStore((state = { snippetData: [], spellData: {} }) => {
  console.log(snippetData);
});
