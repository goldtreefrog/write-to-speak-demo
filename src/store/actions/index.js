export const ADD_SNIPPET = "ADD_SNIPPET";
export const addSnippet = (type, action) => ({
  type,
  snippet: action.snippet
});

export const UPDATE_SNIPPET = "UPDATE_SNIPPET";
export const updateSnippet = (type, action) => ({
  type: UPDATE_SNIPPET,
  snippet: action.snippet
});

export const DELETE_SNIPPET = "DELETE_SNIPPET";
export const deleteSnippet = snippetText => ({
  type: DELETE_SNIPPET,
  snippetText
});

export const SET_SNIPPETS_AVAILABILITY = "SET_SNIPPETS_AVAILABILITY";
export const setSnippetsAvailability = (type, action) => ({
  type: type,
  snippetsAvail: action.snippetsAvail
});

export const WRITING_AREA_VISIBLE = "WRITING_AREA_VISIBLE";
export const writingAreaVisible = () => ({
  type: WRITING_AREA_VISIBLE
});

export const WRITING_AREA_HIDDEN = "WRITING_AREA_HIDDEN";
export const writingAreaHidden = () => ({
  type: WRITING_AREA_HIDDEN
});

export const SPELLING_AREA_VISIBLE = "SPELLING_AREA_VISIBLE";
export const spellingAreaVisible = () => ({
  type: SPELLING_AREA_VISIBLE
});

export const SPELLING_AREA_HIDDEN = "SPELLING_AREA_HIDDEN";
export const spellingAreaHidden = () => ({
  type: SPELLING_AREA_HIDDEN
});

export const IS_EDITING = "IS_EDITING";
export const isEditing = (type, action) => ({
  type: IS_EDITING,
  isEditing: action.isEditing,
  editingPage: action.editingPage
});

export const WRITING_AREA_POPULATE = "WRITING_AREA_POPULATE";
export const writingAreaPopulate = (type, action) => ({
  // type: WRITING_AREA_POPULATE,
  type: type,
  activeSnippetId: action.activeSnippetId,
  activeSnippetText: action.activeSnippetText
});

export const WRITING_AREA_CHANGE = "WRITING_AREA_CHANGE";
export const writingAreaChange = (type, action) => ({
  type,
  activeSnippetText: action.activeSnippetText
});

export const WRITING_AREA_RESET = "WRITING_AREA_RESET";
export const writingAreaReset = () => ({
  type: WRITING_AREA_RESET
});
