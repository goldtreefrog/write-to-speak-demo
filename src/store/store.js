import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { reducer as formReducer } from "redux-form";
import snippetsReducer from "./reducers/snippets-reducer";
import writingReducer from "./reducers/writing-reducer";
import spellingReducer from "./reducers/spelling-reducer";
import otherReducer from "./reducers/other-reducer";
import authReducer from "./reducers/auth.js";
import protectedDataReducer from "./reducers/protected-data.js";

const rootReducer = combineReducers({
  // you have to pass formReducer under 'form' key,
  // for custom keys look up the docs for 'getFormState'
  form: formReducer,
  // ...your other reducers here
  snippets: snippetsReducer,
  writing: writingReducer,
  spelling: spellingReducer,
  other: otherReducer,
  auth: authReducer,
  protectedData: protectedDataReducer
});

// Use to make Redux dev tools work in Chrome. See https://github.com/zalmoxisus/redux-devtools-extension
const composeEnhancers =
  (window.__REDUX_DEVTOOLS_EXTENSION__ &&
    window.__REDUX_DEVTOOLS_EXTENSION__()) ||
  compose;

// const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)));
const store = createStore(
  rootReducer,
  composeEnhancers,
  applyMiddleware(thunk)
);

export default store;
