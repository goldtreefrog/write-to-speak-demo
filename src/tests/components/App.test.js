import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { App } from "./../../components/App.js";
import { store } from "./../../store/store.js";

it("renders without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
