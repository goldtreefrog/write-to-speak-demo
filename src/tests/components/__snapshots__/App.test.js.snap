// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Header />
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/write"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/talk"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </main>
    <Footer />
  </div>
</BrowserRouter>
`;
